### PR TITLE
Fixed display alerts on helpdesk interface

### PR DESCRIPTION
**Pull request by edgardmessias**

### CODE REVIEW

File: inc/alert.class.php [hunk 1]
```diff
@@ -528,7 +531,7 @@ static function preItemForm($params = []) {
          $itemtype    = get_class($item);
          $entities_id = isset($params['item']->fields['entities_id'])
             ? $params['item']->fields['entities_id']
-            : 0;
+            : false; // false to use current entity
```
Comment (cedric-anne) [inline inc/alert.class.php]: OK for this. I pushed it in develop branch (8197caf72eb45f8023b225c5977df86dee33dcad).

File: inc/alert.class.php [hunk 2]
```diff
@@ -428,7 +428,10 @@ public function showForm($ID, $options = []) {
 
    static function displayOnCentral() {
       echo "<tr><th colspan='2'>";
-      self::displayAlerts(['show_only_login_alerts' => false]);
+      self::displayAlerts([
+          'show_only_login_alerts' => false,
+          'show_helpdesk_alerts'   => $_SESSION['glpiactiveprofile']['interface'] == 'helpdesk',
```
[the remark this answers is not in the document]
Comment (edgardmessias) [reply inc/alert.class.php]: Does not display if the user is in a child entity and not recursive.

Example: ROOT -> Alpha.

If user is in Alpha and non recursive and I set alert on ROOT and recursive, dont show to user with helpdesk interface

[the remark this answers is not in the document]
Comment (cedric-anne) [reply inc/alert.class.php]: Problem with your solution is that the option `show_helpdesk_alerts` prevent display of other alerts, so on home page (central page), you do not see alerts made to be displayed only there.

My proposal to fix this and your use case :
```diff
diff --git a/inc/alert.class.php b/inc/alert.class.php
index 9f0dde5..107e458 100644
--- a/inc/alert.class.php
+++ b/inc/alert.class.php
@@ -246,13 +246,9 @@ class PluginNewsAlert extends CommonDBTM {
       //and not the current entity
       if ($p['show_helpdesk_alerts']) {
          $show_helpdesk_sql = " AND `$table`.`is_displayed_onhelpdesk`='1'";
-         $entity_sql = getEntitiesRestrictRequest(" AND", $table, "",
-                                                  $p['entities_id'], true);
-      } else if (!$p['show_only_login_alerts']) {
-         $entity_sql = getEntitiesRestrictRequest("AND", $table, "",
-                                                  $_SESSION['glpiactive_entity'],
-                                                  $_SESSION['glpiactive_entity_recursive'],
-                                                  true);
+      }
+      if (!$p['show_only_login_alerts']) {
+         $entity_sql = getEntitiesRestrictRequest("AND", $table, "", $p['entities_id'], true);
       }
 
       $query = "SELECT DISTINCT `$table`.`id`, `$table`.*
```

**Comment by edgardmessias** [comment]
Updated with your suggestion

**Comment by cedric-anne** [comment]
@orthagh Could you tell us if you approuve this change ?

**Comment by orthagh** [comment]
ok for me

**Comment by cedric-anne** [comment]
Hi,

Thanks for the contribution. It will be added to the next release.

Regards

**Pull request merged**